packages/core/src/render3/errors.ts

```diff
@@ -70,6 +70,25 @@ export function throwErrorIfNoChangesMode(
   throw new RuntimeError(RuntimeErrorCode.EXPRESSION_CHANGED_AFTER_CHECKED, msg);
 }
 
+function formatValue(value: any): unknown {
+  let strValue: string|undefined = value?.toString?.();
+  const stringLengthLimit = 200;
```

Comment:
We can probably move the const out of the fn itself and we could also use `String` to always have a string representation of the value.
```suggestion
/**
 * Configures a max length of a string that represents a value within
 * the `ExpressionChangedAfterItHasBeenCheckedError` error.
 */
const EXPRESSION_CHANGED_VALUE_MAX_LENGTH = 200;

function formatValue(value: unknown): string {
  let strValue = String(value);
  // If it's an array or an object (not a Map, Set, etc), use `JSON.stringify` to improve the error message.
  if (Array.isArray(value) || strValue === '[object Object]') {
    // Use try-catch block, since JSON.stringify throws on circular references
    try {
      strValue = JSON.stringify(value);
    } catch (error) {}
  }
  return strValue.length > EXPRESSION_CHANGED_VALUE_MAX_LENGTH
    ? (strValue.substring(0, stringLengthLimit) + '…')
    : strValue;
```